docs/functionaldatatypes/nonemptylist.md

```diff
@@ -0,0 +1,195 @@
+
+Since we know that a `NonEmptyList` contains at least one value, we don't need to provide an initial value for when the list is empty and can use the `reduceLeft` and `reduceRight` operators. However, these operators don't provide us quite the power we need in a lot of situations because they don't allow the type of the summary value to differ from the type of the elements in the collection.
+
+The `reduceMapLeft` and `reduceMapRight` operators allow us to transform the first value in the `NonEmptyList` to the summary value and then combine each other value with that summary value. This is useful because it allows us to preserve the information that the collection is not empty.
+
+For example, here is how we could use `reduceMapLeft` to convert a `NonEmptyList` to a `NonEmptyChunk`.
+
+```scala mdoc:reset
+import zio.NonEmptyChunk
+import zio.prelude.NonEmptyList
+
+def toNonEmptyChunk[A](as: NonEmptyList[A]): NonEmptyChunk[A] =
+  as.reduceMapLeft(a => NonEmptyChunk(a))(_ appended _)
+```
+
+This would have been difficult to express with `foldLeft` because we would have had to start with an initial value that was an empty chunk and thus would have gotten back a `Chunk` instead of a `NonEmptyChunk`.
+
+Of course we could have just used the `toNonEmptyChunk` operator on `NonEmptyList`.
+
+```scala mdoc:nest
+def toNonEmptyChunk[A](as: NonEmptyList[A]): NonEmptyChunk[A] =
+  as.toNonEmptyChunk
+```
+
+## Interoperability With Scala Collections
+
+One of the great features of the `NonEmptyList` data type is its excellent integration with the Scala standard library collections framework.
+
+A `NonEmptyList` is implicitly convertible into the `::` case of a `List` so anywhere you need a `List` and provide a `NonEmptyList` it will just work.
+
+```scala mdoc:reset
+import zio.prelude.NonEmptyList
+
+def myExistingAPI(as: List[Int]): String =
+  "Some fancy stuff"
+
+val nonEmptyList: NonEmptyList[Int] =
+  NonEmptyList(1, 2, 3)
+
+myExistingAPI(nonEmptyList) // okay
+```
+
+This is also extremely convenient because it means all operators available on `List` are available on `NonEmptyList`.
+
+For example, the `filter` operator is not implemented directly on `NonEmptyList`. But you can still filter a `NonEmptyList` because we can always view a `NonEmptyList` as a `List`.
+
+```scala mdoc
+val filtered: List[Int] =
+  nonEmptyList.filter(_ % 2 == 0) // okay
+```
+
+The return type of `List[Int]` is as specific as possible because the `filter` operator could filter out all the elements of the list and return a list that is empty.
+
+Thus, `NonEmptyList` lets us work in a way that is as close as possible to being as if `NonEmptyList` was a subtype of `List`, which conceptually it is, without us being able to actually rewrite the Scala standard library.
```

Comment:
wording "without us being able to actually rewrite the Scala standard library." is unclear and seems irrelevant. No one expects us to rewrite the Scala standard library. This reads well by skipping that wording entirely.

Reply:
The whole page is excellent and uses compelling context.

Reply:
Good suggestion.